### PR TITLE
multi-item scoring test

### PR DESCRIPTION
**Pull request by EdwardXuy**
[Ascend] Add multi-item scoring tests for NPU basic functionality

Summary:
This commit introduces comprehensive test cases for multi-item scoring
functionality on Ascend NPU, ensuring correct behavior of softmax scoring
and parameter handling in SGLang's Ascend backend.

Test Files Added:

test_npu_multi_item_scoring_basic.py

Validates semantic correctness of multi-item scoring
Verifies that "Yes" token receives higher scores for positive queries
Tests basic scoring pipeline with tokenized input
test_npu_multi_item_scoring_softmax.py

Tests apply_softmax=False vs apply_softmax=True parameter distinction
Verifies softmax probability normalization (scores sum to 1.0)
Covers both tokenized and text input scenarios
test_ascend_utils.py (Helper)

Provides utility functions for Ascend NPU test setup
Includes server management and scoring helper methods
Key Validations:
✓ apply_softmax=False returns raw logits
✓ apply_softmax=True returns normalized probabilities (sum = 1.0)
✓ Correct token ranking for semantic scoring
✓ All tests pass on Ascend 910B with CANN 8.5.0

Test Results:

test_npu_multi_item_scoring_basic.py: 1/1 passed
test_npu_multi_item_scoring_softmax.py: 2/2 passed
Total: 3/3 tests passed
Related:

Improves test coverage for Ascend NPU scoring functionality
Enables CI validation for multi-item scoring parameters

### CODE REVIEW

File: test/registered/ascend/basic_function/parameter/test_npu_multi_item_scoring_basic.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        """
+        response = send_score_request(
+            base_url=self.base_url,
+            query=_QUERY,
+            items=_ITEMS,
+            label_token_ids=_LABEL_TOKEN_IDS,
+            apply_softmax=True,
+        )
+
+
```
Comment (cherryblo) [inline test/registered/ascend/basic_function/parameter/test_npu_multi_item_scoring_basic.py]: delete it

File: test/registered/ascend/basic_function/parameter/test_npu_multi_item_scoring_softmax.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+                msg=(
+                    f"scores[{idx}] must sum to 1.0 with apply_softmax=True "
+                    f"(actual sum: {total:.8f})."
+                ),
+            )
+        logger.info("apply_softmax=True verified: all item score lists sum to 1.0.")
+
+    def test_apply_softmax_false_tokenized_input(self):
+        """
+        Verify apply_softmax=False with pre-tokenized query and items (Case 6 merged):
```
Comment (cherryblo) [inline test/registered/ascend/basic_function/parameter/test_npu_multi_item_scoring_softmax.py]: delete “(Case 6 merged)”

File: test/registered/ascend/basic_function/parameter/test_npu_multi_item_scoring_basic.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        )
+
+        self.assertEqual(
+            response.status_code,
+            200,
+            f"Expected HTTP 200, got {response.status_code}: {response.text}",
+        )
+
+        result = response.json()
+
```
Comment (cherryblo) [inline test/registered/ascend/basic_function/parameter/test_npu_multi_item_scoring_basic.py]: delete it

File: test/registered/ascend/basic_function/parameter/test_npu_multi_item_scoring_softmax.py [hunk 4]
```diff
@@ -0,0 +1,221 @@
+import logging
```
Comment (cherryblo) [inline test/registered/ascend/basic_function/parameter/test_npu_multi_item_scoring_softmax.py:1]: This test case will be merged with the above ones to reduce service startup time.

File: python/sglang/test/ascend/test_ascend_utils.py [hunk 5]
```diff
@@ -549,3 +551,58 @@ async def _run():
...
+            The JSON body contains a "scores" field: list[list[float]],
+            one sub-list per item, each of length len(label_token_ids).
+    """
+
+    payload = {
+        "query": query,
+        "items": items,
+        "label_token_ids": label_token_ids,
+        "apply_softmax": apply_softmax,
+        "item_first": item_first,
```
Comment (cherryblo) [inline python/sglang/test/ascend/test_ascend_utils.py]: This request parameter is not tested.
